meta-mender-core/recipes-bsp/grub-mender-grubenv/grub-mender-grubenv.inc

```diff
@@ -52,20 +52,12 @@ do_configure() {
         mender_rootfsb_part=$(get_part_number_from_device ${MENDER_ROOTFS_PART_B})
     fi
 
-    if [ -n "${MENDER_GRUB_STORAGE_DEVICE}" ]; then
-        mender_grub_storage_device=${MENDER_GRUB_STORAGE_DEVICE}
-    elif ${@bb.utils.contains('DISTRO_FEATURES', 'mender-partuuid', 'true', 'false', d)}; then
-        bbfatal "MENDER_GRUB_STORAGE_DEVICE is required for mender-partuuid"
-    else
-        mender_grub_storage_device=$(get_grub_device_from_device_base ${MENDER_STORAGE_DEVICE_BASE})
-    fi
-
     cat > ${B}/mender_grubenv_defines <<EOF
 mender_rootfsa_part=$mender_rootfsa_part
 mender_rootfsb_part=$mender_rootfsb_part
 mender_kernel_root_base=${MENDER_STORAGE_DEVICE_BASE}
-mender_grub_storage_device=$mender_grub_storage_device
 kernel_imagetype=${KERNEL_IMAGETYPE}
+regexp (.*),(.*) \$root -s mender_grub_storage_device
```

Comment:
I love this patch!!

But do you think it could go as an extra file in [grub-mender-grubenv](https://github.com/mendersoftware/grub-mender-grubenv) instead, say `04_grub_storage_device`? Then it would also benefit mender-convert.

Testing them together with the rest of this patch is a bit of a PITA, but I can help with that.

Reply:
The only concern I have is can we guarantee that the regexp module is available there? I explicitly had to turn it on in Yocto so if we are using distro-provided Grub binaries we should probably take a look to see if regexp is reasonably standard.

Reply:
Good point. I have previously argued that the list of modules should live in grub-mender-grubenv, but we have not had the time to make this happen, so it's duplicated in meta-mender and in mender-convert-integration-scripts ATM. If you could fix *that* I would of course be super happy, but then I'm fine with keeping it here also for the time being! :-)

Reply:
This goes in with https://github.com/mendersoftware/mender-docs/pull/894